### PR TITLE
Remove dependency to assert polyfill

### PR DESCRIPTION
**Pull request by curtisman**
We already switched all of our assert to use our implementation in common-utils in our client packages. 
And node already have it built-in (for server packages)

So we can remove our dependency on the polyfill packages.

### CODE REVIEW

File: server/routerlicious/packages/protocol-base/package.json [hunk 1]
```diff
@@ -57,8 +57,7 @@
     "@fluidframework/common-utils": "^0.31.0",
     "@fluidframework/gitresources": "^0.1028.0",
     "@fluidframework/protocol-definitions": "^0.1024.0",
-    "assert": "^2.0.0",
-    "lodash": "^4.17.21"
+     "lodash": "^4.17.21"
```
Comment (ChumpChief) [inline server/routerlicious/packages/protocol-base/package.json:60]:
```suggestion
    "lodash": "^4.17.21"
```

File: packages/utils/odsp-doclib-utils/src/test/tsconfig.json [hunk 2]
```diff
@@ -4,6 +4,7 @@
         "rootDir": "./",
         "outDir": "../../dist/test",
         "types": [
+            "node",
```
Comment (ChumpChief) [inline packages/utils/odsp-doclib-utils/src/test/tsconfig.json:7]: Is this addition intentional?  Looks like you're adding node types in 6 packages in this change.

**Review by ChumpChief — APPROVED** [review]

File: packages/utils/odsp-doclib-utils/src/test/tsconfig.json (hunk 2 above)
Comment (ChumpChief) [reply packages/utils/odsp-doclib-utils/src/test/tsconfig.json:7]: Ah I see it's because they're all test configs, never mind.

Comment (curtisman) [reply packages/utils/odsp-doclib-utils/src/test/tsconfig.json:7]: Yes, these are all configs for tests only (separate from main packages), and they are only ran in node.
